modules/tide_edit_protection/tide_edit_protection.module

```diff
@@ -0,0 +1,129 @@
+ * Contains tide_edit_protection.module.
+ */
+
+use Drupal\Component\Utility\Html;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\views\ViewExecutable;
+
+/**
+ * Implements hook_help().
+ */
+function tide_edit_protection_help($route_name, RouteMatchInterface $route_match) {
+  switch ($route_name) {
+    case 'help.page.tide_edit_protection':
+      $output = '';
+      $output .= '<h3>' . t('About') . '</h3>';
+      $output .= '<p>' . t('Protects unsaved change.') . '</p>';
+      return $output;
+
+    default:
+  }
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function tide_edit_protection_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $include = FALSE;
+  $config = \Drupal::config('tide_edit_protection.form');
+
+  // Check entity form conditions.
+  $entity_method = $config->get('entity_method');
+  if (!empty($entity_method)) {
+    $formObject = $form_state->getFormObject();
+    if (method_exists($formObject, 'getEntity')) {
+      $entity = $formObject->getEntity();
+      $entity_type = $entity->getEntityTypeId();
+      $entity_bundle = $entity_type . '__' . $entity->bundle();
+      if ('all' === $entity_method) {
+        $include = TRUE;
+      }
+      elseif ('entity_type' === $entity_method) {
+        $entity_types = $config->get('entity_types');
+        if (in_array($entity_type, $entity_types)) {
+          $include = TRUE;
+        }
+      }
+      elseif ('entity_bundle' === $entity_method) {
+        $entity_bundles = $config->get('entity_bundles');
+        if (in_array($entity_bundle, $entity_bundles)) {
+          $include = TRUE;
+        }
+      }
+    }
+  }
+
+  // Check other forms by form_id, support Regex.
+  if (FALSE === $include) {
+    $alert_form_ids = $config->get('alert_form_ids');
+    if (!empty($alert_form_ids)) {
+      $alert_form_ids = explode("\n", $alert_form_ids);
+      foreach ($alert_form_ids as $alert_form_id) {
+        if (preg_match('/^[a-zA-Z_-]$/', $alert_form_id)) {
+          if ($alert_form_id === $form_id) {
+            $include = TRUE;
+            break;
+          }
+        }
+        else {
+          if (preg_match('/' . $alert_form_id . '/i', $form_id)) {
+            $include = TRUE;
+            break;
+          }
+        }
+      }
+    }
+  }
+
+  if (TRUE === $include) {
+    $cleaned_id = Html::getId($form_id);
+    $form['#attached']['library'][] = 'tide_edit_protection/form_edit_protection';
+    $form['#attached']['drupalSettings']['tide_edit_protection']['forms'][$cleaned_id] = $cleaned_id;
+  }
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function tide_edit_protection_form_views_exposed_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  if (\Drupal::moduleHandler()->moduleExists('tide_site_restriction')) {
+    $form_ids = [
+      'views-exposed-form-tide-locked-content-page-1' => 'uid_current',
+    ];
+    foreach ($form_ids as $form_id => $filter) {
```

Comment:
Why is a loop necessary if the array that's being passed to it is singular?

Reply:
Thanks @GROwen , yes. removed the loop.